purchases/src/test/java/com/revenuecat/purchases/PurchasesCoroutinesTest.kt

```diff
@@ -94,4 +95,93 @@ class PurchasesCoroutinesTest : BasePurchasesTest() {
     }
 
     // endregion
+
+    // region awaitOfferings
+    @Test
+    fun `retrieve offerings - Success`() = runTest {
+        mockOfferingsManagerGetOfferings()
+
+        val result = purchases.awaitOfferings()
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNotNull
+    }
+
+    @Test
+    fun `retrieve offerings - Success - offerings match expectations`() = runTest {
+        val offerings = mockOfferingsManagerGetOfferings()
+
+        val result = purchases.awaitOfferings()
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNotNull
+        assertThat(result).isEqualTo(offerings)
+    }
+
+    @Test
+    fun `retrieve offerings - Error`() = runTest {
+        val error = PurchasesError(PurchasesErrorCode.UnknownError, "Offerings error")
+        mockOfferingsManagerGetOfferings(error)
+
+        var result: Offerings? = null
+        var exception: Throwable? = null
+        runCatching {
+            result = purchases.awaitOfferings()
+        }.onFailure {
+            exception = it
+        }
+
+        verify(exactly = 1) {
+            mockOfferingsManager.getOfferings(
+                appUserId,
+                any(),
+                any(),
+                any(),
+            )
+        }
+        assertThat(result).isNull()
+        assertThat(exception).isNotNull
+    }
+
+    @Test
+    fun `retrieve offerings - ConfigurationError`() = runTest {
+        val error = PurchasesError(PurchasesErrorCode.ConfigurationError, "Offerings config error")
+        mockOfferingsManagerGetOfferings(error)
+
+        var result: Offerings? = null
+        var exception: Throwable? = null
+        runCatching {
+            result = purchases.awaitOfferings()
```

Comment:
Alternatively, we could add a `fail` after this line... I kinda prefer it since `result` is unused here. Thoughts?

Reply:
The only problem with using `fail` is that the catch will catch the `fail` and store it as an exception. So this test will fail checking the type of the exception, not because the `fail` throws an exception

Reply:
Ohhh yeah, that makes sense... I guess ideally we would only catch the type of exception we are expecting, so others would just make it fail the test as well... But I'm not sure how that would look like here. This is ok then!